packages/offix-datastore/src/storage/LocalStorage.ts

```diff
@@ -17,15 +17,45 @@ export function generateId() {
 export class LocalStorage {
   public readonly storeChangeEventStream: PushStream<StoreChangeEvent>;
   public readonly adapter: StorageAdapter;
+  private eventQueue: StoreChangeEvent[] = [];
 
-  constructor(adapter: StorageAdapter) {
-    this.storeChangeEventStream = new ObservablePushStream();
+  constructor(adapter: StorageAdapter, pushStream?: PushStream<StoreChangeEvent>) {
+    this.storeChangeEventStream = pushStream || new ObservablePushStream();
     this.adapter = adapter;
   }
 
+  /**
+   * @returns a new LocalStorage instance in transaction mode
+   * In transaction, no events are fired unitll the transaction is committed.
+   */
+  public async createTransaction() {
+    const adapterTransaction = await this.adapter.createTransaction();
+    return new LocalStorage(adapterTransaction, this.storeChangeEventStream);
+  }
+
+  /**
+   * Commits a transaction if one is open.
+   * It fires all events that occured in this transaction
+   */
+  public async commit() {
+    if (!this.adapter.isTransactionOpen()) { return; }
```

Comment:
This should throw really ugly error! 
Why. If this ever happens it usually will mean bug as you should not have code that commits transition that were not opened.